### DISCUSSION TITLE
Is it possible to obtain the token variable or user session inside redirect callback ?

### DISCUSSION THREAD

**Discussion by weidezhang** [nextauthjs/next-auth#8950] [Help]
### Summary

My purpose is to redirect user to a different page based on user attribute (from user object or token or popupated session) calledd user type.  Is it possible to achieve it inside redirect callback ?  I'm not sure how to get the user data in redirect callback. 

Currently the signature for redirect callbcak in next_auth api is defined like the following: 
...
callbacks: {
  async redirect({ url, baseUrl }) {
    // Allows relative callback URLs
    if (url.startsWith("/")) return `${baseUrl}${url}`
    // Allows callback URLs on the same origin
    else if (new URL(url).origin === baseUrl) return url
    return baseUrl
  }
}
...

**Comment by weidezhang** [comment]
found the answer. thx

**Reply by MatteoGauthier** [reply]
And what's the answer ?

**Reply by hozaifa4you** [reply]
> found the answer. thx

Where it is?
